src/lib.rs

```diff
@@ -419,6 +426,8 @@ pub struct WindowAttributes {
     /// Whether the window should be set as fullscreen upon creation.
     ///
     /// The default is `None`.
+    #[cfg_attr(feature = "serde", doc("\nThis field does not currently get serialized or deserialized."))]
+    #[cfg_attr(feature = "serde", serde(skip))] // TODO: FIGURE OUT HOW TO SERIALIZE THIS
     pub fullscreen: Option<MonitorId>,
```

Comment:
I'm not sure this can be serialized in any meaningful way. This would require that the monitors have a consistent ids as seen by the app between two executions. Do any platform provide this kind of guarantee ?

Reply:
I can't say I know enough about the platform-specific APIs to give this a firm answer. However, I'd be really surprised if they didn't - being able to enumerate over a consistent list of monitors and choose the right one is really important on multi monitor setups, especially since the first monitor reported isn't always the one users want your application to be on.

Since you're brining this up, does Wayland not provide APIs to do this consistently?

Reply:
To be honest I'm not sure.

- On the one hand there are the global ids, that uniquely identify a monitor, but have no guarantee to be stable across reboots, not even to still be valid at all.
- On the other hand, the wayland server reports the "name" of the monitor (in the form of model an make), which is stable across reboots, but are not unique identifiers (two monitors of the same model would have the same name I believe).

Reply:
Hmm. Is the order in which displays are reported consistent across reboots? If that's the case, we could see about serializing an index into the reported display list, although that would be rather hacky.

If we can't find a good solution for this soonish, I'd say that we should make a separate issue for figuring out how to proceed and then merge this as-is.

Reply:
I think leaving this as a separate issue is probably a good approach here.

Reply:
I'm okay with that.